### ISSUE TITLE
Ticker process interferes with other processes

### ISSUE THREAD

**Issue by TenzinCHW** [nlsynth/karuta#10]
Hi, I am running Ubuntu 20.04 with karuta-0.6.1 I do not know if this is a bug, but when I try to create a slow clock process as specified in the docs in the same file as another process, only the first one actually runs. E.g.
```
shared ticker object = Env.newTicker()
shared slowclk #0 = 0 

process slow_clk_proc() {
    ticker.decrementCount(ticker.getCount())
    print("slowclk enter")
    while true {
        print("slowclk always")
        wait(60 - ticker.getCount())
        ticker.decrementCount(60)
        slowclk = ~slowclk
    }   
}

process sd_card() {
    print("sdcard enter")
    while true {
        print("sdcard always")
    }   
}
```

will give me the following output when I run `karuta run --duration 10 testclk.karuta`
```
print: slowclk enter
print: sdcard enter
print: slowclk always
print: slowclk always
print: slowclk always
print: slowclk always
print: slowclk always
```

And if I swap the order:

```
shared ticker object = Env.newTicker()
shared slowclk #0 = 0 

process sd_card() {
    print("sdcard enter")
    while true {
        print("sdcard always")
    }   
}
 
process slow_clk_proc() {
    ticker.decrementCount(ticker.getCount())
    print("slowclk enter")
    while true {
        print("slowclk always")
        wait(60 - ticker.getCount())
        ticker.decrementCount(60)
        slowclk = ~slowclk
    }   
}
```

I get instead

```
print: sdcard enter
print: slowclk enter
print: sdcard always
print: sdcard always
print: sdcard always
print: sdcard always
print: sdcard always
```

My interval is 60 clocks so I tried different durations like 100 or 1000 but the pattern doesn't change.

**Comment by yt76** [comment]
Thank you for reporting this issue.
I released Karuta 0.6.2 with a fix of scheduling fairness. So this issue should be resolved.

One possible caveat is that Karuta's interpreter mode doesn't have a global clock counter and doesn't care how much time each thread run. You will see (almost) same number of "sdcard always" and "slowclk always".

**Comment by TenzinCHW** [comment]
Alright thank you for the speedy fix!

If there's a problem with synthesis I will reopen this issue.
